### DISCUSSION TITLE
Is it possible to rerun parent loader function or child route change?

### DISCUSSION THREAD

**Discussion by lukejagodzinski** [remix-run/remix#3241] [Q&A] [answered]
Hi, I have the following routes structures:

```
.
└── routes/
    ├── $company/
    │   ├── $prompts/
    │   │   ├── index.tsx
    │   │   └── $promptName.tsx
    │   ├── index.tsx
    │   └── $prompts.tsx
    ├── $company.tsx
    └── index.tsx
```

So an example routes between which I navigate are:
`/companyName/prompts/a`
`/companyName/prompts/b`
`/companyName/prompts/c`
And everything works fine. Whenever I navigate, the `loader` function gets called for each prompt. However I would also like to display some data based on the selected prompt at the higher level. I would like to display different thing in the `$company` file based on the `promptName` param in the URL but that never happens. The loader function only gets rerun for the `$company` route when the `company` param changes, so the `useLoaderData` function never gets new data.

Is there a way to force it to reload given loader?

I guess I would have to do something like:

```
const fetcher = useFetcher();
const params = useParams();
useEffect(() => {
  fetcher.load(/* ... */);
}, [params]);
```

But I would like to avoid doing that.

**Comment by kiliman** [comment, marked as answer]
Remix optimizes client navigation by only calling the loaders of the changed routes. If you want to render something different on a parent route based on child data, then return the data from the child loader and access it in your parent component via `useMatches`
```ts
// child.tsx
export function loader() {
  return json(dataForParent)
}

// parent.tsx
export default function() {
  const loaderData = useLoaderData() // parent data
  const childData = useMatches().at(-1).data // get route data from last (child) route
}
```

**Reply by lukejagodzinski** [reply]
Oh thank you! That's what I was looking for. Tbh. from documentation I would never guess that it does what it does.

**Reply by kiliman** [reply]
You'll find that `useMatches` + `handle` is the Swiss Army Knife of Remix.
